docs/modules/ROOT/pages/authenticationclass.adoc

```diff
@@ -1,3 +1,48 @@
 = `AuthenticationClass`
 
-TODO
+`AuthenticationClass` is a CRD describing a generic authentication method like LDAP or Kerberos.
+Multiple operators use this CRD as a way to express the authentication of the product.
+
+At the moment the following Authentication providers are supported:
+
+* <<LDAP>>
+
+== LDAP
+A very simple `AuthenticationClass` with LDAP Authentication looks like this:
+
+[source,yaml]
+----
+include::example$authenticationclass-ldap-simple.yaml[]
+----
+<1> The hostname of the LDAP server without any protocol or port
+<2> The port of the LDAP server. If TLS is used it defaults to `636` otherwise to `389`
+<3> An optional searchBase where the users should be searched
+
+=== OpenLDAP
+Here is an example that is tuned for an OpenLDAP LDAP server:
+
+[source,yaml]
+----
+include::example$authenticationclass-ldap-openldap-simple.yaml[]
+----
+<1> The name of the xref:secret-operator::secretclass.adoc[] providing the bind credentials (username and password). Must match the name of the xref:secret-operator::secretclass.adoc[] in this example in ②
+<2> The name of the xref:secret-operator::secretclass.adoc[]
```

Comment:
I had to think a bit to understand this, maybe add something like

"The name of the SecretClass that is referred to by (1)"  or similar